### PR TITLE
[ETP-460] [WIP] JS cleanup. Tickets block, modal, data libraries.

### PR DESCRIPTION
**Pull request by juanfra**
🎫 ETP-460

🗒️ This is a WIP so the team can see how are things looking while I'm on vacation.

### CODE REVIEW

File: src/Tribe/Editor/Blocks/Tickets.php [hunk 1]
```diff
@@ -143,6 +148,43 @@ public function assets() {
...
+		if ( tribe_tickets_new_views_is_enabled() ) {
+			// @todo: Include only when ET+ is active and modal is there.
+			tribe_asset(
+				$plugin,
+				'tribe-tickets-modal',
+				'v2/tickets-modal.js',
+				[
+					'jquery',
+					'tribe-common',
+					//'tribe-tickets-block',
```
Comment (tr1b0t) [inline src/Tribe/Editor/Blocks/Tickets.php:160]: :no_entry_sign: No space found before comment text; expected "// 'tribe-tickets-block'," but found "//'tribe-tickets-block',"

<sub>hash: 4011d8de36fc8bf46bd13032e4267a0e</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 1 error(s)

File: src/resources/js/v2/tickets-data.js [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+					ret = {
+						meta: data.meta,
+						tickets: data.tickets,
+					};
+
+					deferred.resolve( ret );
+				},
+				error: function() {
+					deferred.reject( false );
+				},
```
Comment (bordoni) [inline src/resources/js/v2/tickets-data.js:248]: Lets move into separate methods.

File: src/resources/js/v2/tickets-modal.js [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+
+						for ( let i = counter; i <= qty; i++ ) {
+							const data = { attendee_id: i };
+
+							$ticketContainer.append( ticketTemplate( data ) );
+							obj.maybeHydrateAttendeeBlockFromLocal( $existing.length );
+						}
+					}
+				}
+			}
```
Comment (bordoni) [inline src/resources/js/v2/tickets-modal.js:117]: Separate method.

File: src/resources/js/v2/tickets-modal.js [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+							if ( value === $item.val() ) {
+								$item.prop( 'checked', true );
+							}
+						} );
+					}
+				} );
+
+				current++;
+			} );
+		} );
```
Comment (bordoni) [inline src/resources/js/v2/tickets-modal.js:172]: Separate methods.

File: src/resources/js/v2/tickets-modal.js [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+	 * @since TBD
+	 */
+	obj.initModalFormPreFills = function() {
+		// @todo: Fix how this is handling stuff (use container based).
+		const $tribeTicket = $form.closest( tribe.tickets.block.selectors.container );
+
+
+		tribe.tickets.loader.show( $document );
+		$.when(
+			obj.getData()
```
Comment (bordoni) [inline src/resources/js/v2/tickets-modal.js:220]: Missing anon function here.

File: src/resources/js/v2/tickets-modal.js [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+						tribe.tickets.block.updateItem( id, $modalCartItem, $blockCartItem );
+					}
+				);
+
+				obj.initModalFormPreFills();
+
+				tribe.tickets.block.updateFormTotals( $modalCart );
+
+				tribe.tickets.loader.hide( $form );
+			}
```
Comment (bordoni) [inline src/resources/js/v2/tickets-modal.js:355]: Let's make it into a separate method.

File: src/resources/js/v2/tickets-modal.js [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+	/**
+	 * Attempts to hydrate a dynamically-created attendee form "block" from sessionStorage data.
+	 *
+	 * @since TBD
+	 *
+	 * @param {number} length The "skip" index.
+	 */
+	obj.maybeHydrateAttendeeBlockFromLocal = function( length ) {
+		$.when(
+			tribe.tickets.data.getData()
```
Comment (bordoni) [inline src/resources/js/v2/tickets-modal.js:261]: Do we need a function here?

File: src/resources/js/v2/tickets-modal.js [hunk 8]
```diff
@@ -0,0 +1,362 @@
...
+						const $this = $( this );
+						const name = $this.attr( 'name' );
+						const storedVal = data[ name ];
+
+						if ( storedVal ) {
+							$this.val( storedVal );
+						}
+					}
+				);
+			}
```
Comment (bordoni) [inline src/resources/js/v2/tickets-modal.js:291]: Lets use a separate method here.

**Review by bordoni — CHANGES_REQUESTED** [review]
Let's separate some of those Anon function into Object methods as much as possible.

File: src/resources/js/v2/tickets-data.js [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+								value = '';
+							}
+						}
+
+						data[ name ] = value;
+					}
+				);
+
+				tempMeta[ ticketId ].items.push( data );
+			}
```
[the remark this answers is not in the document]
Comment (juanfra) [reply src/resources/js/v2/tickets-data.js:124]: This is not ready for review, pushed so Scott could know where we were at.

File: src/resources/js/v2/tickets-data.js (hunk 2 above)
Comment (juanfra) [reply src/resources/js/v2/tickets-data.js:248]: This is not ready for review, pushed so Scott could know where we were at.

File: src/resources/js/v2/tickets-modal.js (hunk 3 above)
Comment (juanfra) [reply src/resources/js/v2/tickets-modal.js:117]: This is not ready for review, pushed so Scott could know where we were at.

File: src/resources/js/v2/tickets-modal.js (hunk 4 above)
Comment (juanfra) [reply src/resources/js/v2/tickets-modal.js:172]: This is not ready for review, pushed so Scott could know where we were at.

File: src/resources/js/v2/tickets-modal.js (hunk 5 above)
Comment (juanfra) [reply src/resources/js/v2/tickets-modal.js:220]: This is not ready for review, pushed so Scott could know where we were at.

File: src/resources/js/v2/tickets-modal.js (hunk 6 above)
Comment (juanfra) [reply src/resources/js/v2/tickets-modal.js:355]: This is not ready for review, pushed so Scott could know where we were at.

File: src/resources/js/v2/tickets-modal.js (hunk 7 above)
Comment (juanfra) [reply src/resources/js/v2/tickets-modal.js:261]: This is not ready for review, pushed so Scott could know where we were at.

File: src/resources/js/v2/tickets-modal.js (hunk 8 above)
Comment (juanfra) [reply src/resources/js/v2/tickets-modal.js:291]: This is not ready for review, pushed so Scott could know where we were at.

**Comment by juanfra** [comment]
@bordoni thanks for the comments, I'll close this one as we are moving part of the premium functionality to ET+, where it belongs. And I'll be trying to implement a new way of PRing after discussing with Scott (He mentioned that you already talked about it), that I should be pushing things against the "original" version, so it's easier to view what's that we're inheriting.

**Pull request closed without merging**